pkg/server/server_entrypoint_tcp.go

```diff
@@ -616,6 +618,21 @@ func createHTTPServer(ctx context.Context, ln net.Listener, configuration *stati
 
 	prevConnContext := serverHTTP.ConnContext
 	serverHTTP.ConnContext = func(ctx context.Context, c net.Conn) context.Context {
+		// This adds the remote address of the server making the connection if the PROXY Protocol is enabled
```

Comment:
I am wondering if it is better to use this code instead:

```go
	// This adds the remote address of the server making the connection if the PROXY Protocol is enabled
		getProxyIP := func(c net.Conn) string {
			for {
				switch conn := c.(type) {
				case *tcprouter.Conn:
					c = conn.WriteCloser
				case *trackedConnection:
					c = conn.WriteCloser
				case *writeCloserWrapper:
					c = conn.writeCloser
				case *net.TCPConn:
					return c.RemoteAddr().String()
				default:
					return ""
				}
			}
		}

		if proxyProtocolEnabled {
			proxyIP := getProxyIP(c)
			ctx = context.WithValue(ctx, forwardedheaders.ProxyAddrKey, proxyIP)
		}
```

It looks less brittle since it does not depends on the exact order of the "wrapping".  What do you think @juliens, @mmatur ?

Reply:
I am going with this variant!